project/iterative_sorting.py

```diff
@@ -1,36 +1,43 @@
 # Complete the selection_sort() function below in class with your instructor
-def selection_sort( arr ):
+
+
+def selection_sort(arr):
     # loop through n-1 elements
     for i in range(0, len(arr) - 1):
         cur_index = i
         smallest_index = cur_index
         # TO-DO: find next smallest element
-        # (hint, can do in 3 loc) 
-             
-
-
-
-        # TO-DO: swap
-
-
-
-
+        # # (hint, can do in 3 loc)
+        for j in range(cur_index, len(arr)):
+            if arr[j] < arr[smallest_index]:
+                smallest_index = j
+                # TO-DO: swap
+        arr[cur_index], arr[smallest_index] = arr[smallest_index], arr[cur_index]
     return arr
 
-
 # TO-DO: implement the Insertion Sort function below
-def insertion_sort( arr ):
 
+
+def insertion_sort(arr):
+    for i in range(1, len(arr)):
+        cur_index = i
+        while cur_index > 0 and arr[cur_index] < arr[cur_index - 1]:
+            temp = arr[cur_index - 1]
+            arr[cur_index - 1] = arr[cur_index]
+            arr[cur_index] = temp
+            cur_index -= 1
     return arr
```

Comment:
I'm happy to say you got this one working, too. Maybe today you can work on quicksort in the recursive file. 😄